Add REST backend

fd0 opened the pull request:
This is a port of the original work by @bchapuis in
https://github.com/restic/restic/pull/253

TODO:
- [x] Improve documentation

fd0 committed:
Add REST backend

This is a port of the original work by @bchapuis in
https://github.com/restic/restic/pull/253

fd0 committed:
.gitignore: Add /vendor/pkg

fawick commented on doc/REST_backend.md:
```diff
@@ -0,0 +1,56 @@
+Returns "200 OK" if the configuration of the request body has been saved,
+an HTTP error otherwise.
+
+## GET /{type}/
```
The doc should list what would be acceptable values for {type}.

fd0 replied:
Good point.

fd0 committed:
Add rest backend to location

fawick commented on src/restic/backend/rest/backend_test.go:
```diff
@@ -0,0 +1,87 @@
+// DO NOT EDIT, AUTOMATICALLY GENERATED
```
Should the source and the generation recipe for the auto-generation be part of the repo?

fd0 replied:
Oh, it is. Here you can find the go generate comment: https://github.com/restic/restic/pull/464/files#diff-6397b881264cb4356775a0e5a3be197fR15

The file is generated from the file https://github.com/restic/restic/blob/master/src/restic/backend/test/tests.go by the program https://github.com/restic/restic/blob/master/src/restic/backend/test/generate_backend_tests.go

fd0 committed:
Add rest backend to ui parser

fd0 committed:
backend tests: Test accessing config

This commit adds real testing for accessing the config file with
different names.

fd0 committed:
rest backend: Fixes

fd0 commented:
This works with my toy Ruby server and passes all the tests: https://gist.github.com/07c8a0cda34e257565e9

fd0 committed:
rest backend: Improve documentation
